merkle/map_verifier.go

```diff
@@ -40,9 +40,6 @@ func VerifyMapInclusionProof(index []byte, leafHash []byte, expectedRoot []byte,
 	if got, want := len(leafHash)*8, hBits; got != want {
 		return fmt.Errorf("invalid leafHash length %d, expected %d", got, want)
 	}
-	if got, want := len(expectedRoot)*8, hBits; got != want {
```

Comment:
Removing this check allows tests to supply clean dummy values. 
It doesn't reduce the robustness of the checks because the `expectedRoot` is `bytes.Equal` compared with the computed roots.